src/precice/tests/ParallelTests.cpp

```diff
@@ -1104,6 +1105,936 @@ BOOST_AUTO_TEST_CASE(UserDefinedMPICommunicatorPetRBF)
+      BOOST_TEST(!cplInterface.isCouplingOngoing(), "Receiving participant should have to advance once!");
+      cplInterface.finalize();
+
+    } else {
+      int vA = cplInterface.setMeshVertex(meshID, coordOneA.data());
+      int vC = cplInterface.setMeshVertex(meshID, coordOneC.data());
+      int vD = cplInterface.setMeshVertex(meshID, coordOneD.data());
+
+      int eAD = cplInterface.setMeshEdge(meshID, vA, vD);
+      int eCD = cplInterface.setMeshEdge(meshID, vC, vD);
+      int eAC = cplInterface.setMeshEdge(meshID, vA, vC);
+
+      cplInterface.setMeshTriangle(meshID, eAD, eCD, eAC);
+
+      double maxDt = cplInterface.initialize();
+
+      cplInterface.writeScalarData(dataID, vA, valOneA);
+      cplInterface.writeScalarData(dataID, vC, valOneC);
+      cplInterface.writeScalarData(dataID, vD, valOneD);
+
+      cplInterface.advance(maxDt);
+      BOOST_TEST(!cplInterface.isCouplingOngoing(), "Receiving participant should have to advance once!");
+      cplInterface.finalize();
+    }
+
+  } else {
+    SolverInterface cplInterface(context.name, configFilename, context.rank, context.size);
+    const int       meshID = cplInterface.getMeshID("MeshTwo");
+    const int       dataID = cplInterface.getDataID("DataOne", meshID);
+
+    int vA = cplInterface.setMeshVertex(meshID, coordTwoA.data());
+    int vB = cplInterface.setMeshVertex(meshID, coordTwoB.data());
+    int vC = cplInterface.setMeshVertex(meshID, coordTwoC.data());
+    int vD = cplInterface.setMeshVertex(meshID, coordTwoD.data());
+
+    int eAB = cplInterface.setMeshEdge(meshID, vA, vB);
+    int eBC = cplInterface.setMeshEdge(meshID, vB, vC);
+    int eAC = cplInterface.setMeshEdge(meshID, vA, vC);
+    int eCD = cplInterface.setMeshEdge(meshID, vC, vD);
+    int eAD = cplInterface.setMeshEdge(meshID, vA, vD);
+
+    cplInterface.setMeshTriangle(meshID, eAB, eBC, eAC);
+    cplInterface.setMeshTriangle(meshID, eAD, eAC, eCD);
+
+    double maxDt = cplInterface.initialize();
+
+    double valueA, valueB, valueC, valueD;
+    cplInterface.readScalarData(dataID, vA, valueA);
+    cplInterface.readScalarData(dataID, vB, valueB);
+    cplInterface.readScalarData(dataID, vC, valueC);
+    cplInterface.readScalarData(dataID, vD, valueD);
+
+    double calculatedIntegral = (math::geometry::triangleArea(coordTwoA, coordTwoB, coordTwoC) * (valueA + valueB + valueC) +
+                                 math::geometry::triangleArea(coordTwoA, coordTwoC, coordTwoD) * (valueA + valueC + valueD)) /
+                                3.0;
+
+    BOOST_TEST(expectedIntegral == calculatedIntegral);
+
+    cplInterface.advance(maxDt);
+    BOOST_TEST(!cplInterface.isCouplingOngoing(), "Sending participant should have to advance once!");
+    cplInterface.finalize();
+  }
+}
+
+// Simple 2 triangles. Read mapping on parallel participant B.
+// This test does not pass. When the from-mesh is repartitioned, ranks does not own all the vertices in the corresponding triangles.
```

Comment:
But the CI passes?

Reply:
The description was not updated, sorry about that.